### ISSUE TITLE
TSQL: Fetch Cursor Issue.

### ISSUE THREAD

**Issue by regisschmitz** [sqlfluff/sqlfluff#6011]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was having issues using lint with  T-SQL cursors on SQLSERVER where the whole procedure was returning:
`Found unparsable section: 'create procedure [sp_my_procedure]`

### Expected Behaviour

Accept the below _fetch_ statement as valid:
```TSQL
FETCH cursor_name INTO @var1, @var2
```
and list all the issues found in "sp_my_procedure".

### Observed Behaviour

Though is running on sqlserver, sp_my_procedure has sybase legacy syntax.
Then I found the fetch syntax is not as expected on `dialect_tsql.sql`

then I changed the dialect file, to allow "FROM" to be optional too.

```python
match_grammar: Matchable = Sequence(
        "FETCH",
        OneOf("NEXT", "PRIOR", "FIRST", "LAST", optional=True),
        "FROM",
        Ref("CursorNameGrammar"),
        Sequence("INTO", Delimited(Ref("ParameterNameSegment")), optional=True),
    )
```

NEW:

```python
match_grammar: Matchable = Sequence(
        "FETCH",
        OneOf("NEXT", "PRIOR", "FIRST", "LAST", optional=True),
        OneOf("FROM", optional=True), 
        Ref("CursorNameGrammar"),
        Sequence("INTO", Delimited(Ref("ParameterNameSegment")), optional=True),
    )
```
After that, it works successfully with both syntax.

![image](https://github.com/user-attachments/assets/b67641d2-e882-42b3-81c1-f482ad14c878)

### How to reproduce

Any procedure with a cursor that use sybase `fetch` syntax (which sqlserver is compatible)

instead of 
```TSQL
FETCH NEXT FROM cursor_name INTO @var1, @var2
```
use
```TSQL
FETCH cursor_name INTO @var1, @var2
```

### Dialect

tsql

### Version

Python 3.11.3
sqlfluff 3.1.0

### Configuration

[sqlfluff]
dialect = tsql
templater=jinja

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower
unquoted_identifiers_policy = aliases

[sqlfluff:rules:references.quoting]
# Policy on quoted and unquoted identifiers
#['all', 'aliases', 'column_aliases', 'none']
quoted_identifiers_policy = none
prefer_quoted_identifiers = False
prefer_quoted_keywords = True
ignore_words = dbo
ignore_words_regex = None
force_enable = False

[sqlfluff:indentation]
tab_space_size = 3
indent_unit = space
indented_joins = False
indented_ctes = False

indented_using_on = True
indented_on_contents = True
indented_then = True
indented_then_contents = True
allow_implicit_indents = False
template_blocks_indent = True
# This is a comma separated list of elements to skip
# indentation edits to.
skip_indentation_in = script_content
# If comments are found at the end of long lines, we default to moving
# them to the line _before_ their current location as the convention is
# that a comment precedes the line it describes. However if you prefer
# comments moved _after_, this configuration setting can be set to "after".
trailing_comments = before
# To exclude comment lines from indentation entirely set this to "True".
ignore_comment_lines = False

[sqlfluff:rules:references.keywords]
prefer_quoted_identifiers = True
prefer_quoted_keywords = True
ignore_words = dbo
ignore_words_regex = None
force_enable = False

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by griffithska** [comment]
Just ran into this and tested the suggested change to dialect_tsql.py locally and it did resolve my parsing error. Just want to add my support as an end-user.

**Comment by regisschmitz** [comment]
> Just ran into this and tested the suggested change to dialect_tsql.py locally and it did resolve my parsing error. Just want to add my support as an end-user.

@griffithska , tks!
Still waiting the code review and hoping someone to merge ou replicate the change, I am also depending on this change to be able to update the library.  

Best regards!

**Comment by keraion** [comment]
@regisschmitz are you still willing to submit a PR for this change? If so, I do have one suggestion for your fix:

```python
match_grammar: Matchable = Sequence(
        "FETCH",
        OneOf("NEXT", "PRIOR", "FIRST", "LAST", optional=True),
        Ref.keyword("FROM", optional=True), 
        Ref("CursorNameGrammar"),
        Sequence("INTO", Delimited(Ref("ParameterNameSegment")), optional=True),
    )
```

Feel free to tag me in a review if you are still willing to contribute this!

**Comment by regisschmitz** [comment]
Hi @keraion, have you tried this way?
I think I've tried but didn't work.

anyway, it was so long ago that I found several other serious problems that made me abandon sqlfluff.

If you tried and worked perhaps you can make a new pull request.

Best regards.

> [@regisschmitz](https://github.com/regisschmitz) are you still willing to submit a PR for this change? If so, I do have one suggestion for your fix:
> 
> match_grammar: Matchable = Sequence(
>         "FETCH",
>         OneOf("NEXT", "PRIOR", "FIRST", "LAST", optional=True),
>         Ref.keyword("FROM", optional=True), 
>         Ref("CursorNameGrammar"),
>         Sequence("INTO", Delimited(Ref("ParameterNameSegment")), optional=True),
>     )
> 
> Feel free to tag me in a review if you are still willing to contribute this!

**Comment by regisschmitz** [comment]
Hey all, Is this project abandoned?
the pull request is 1 year old already.
